src/components/pages/events/upcoming-events/upcoming-events.jsx

```diff
@@ -10,27 +10,39 @@ import styles from './upcoming-events.module.scss';
 
 const cx = classNames.bind(styles);
 
-const UpcomingEvents = ({ title, upcomingEvents }) => (
+const UpcomingEvents = ({ title, featuredUpcomingEvents, upcomingEventsByYear }) => (
   <section className={cx('wrapper')}>
     <div className="container">
       <Heading className={cx('title')}>{title}</Heading>
       <ul className={cx('items-wrapper')}>
         <div className={cx('cards-wrapper')}>
-          {upcomingEvents.slice(0, 3).map((upcomingEvent, index) => (
+          {featuredUpcomingEvents.map((upcomingEvent, index) => (
             <CardItem key={index} {...upcomingEvent} />
           ))}
         </div>
-        {upcomingEvents.slice(3).map((upcomingEvent, index) => (
-          <Item key={index} {...upcomingEvent} />
-        ))}
+        {upcomingEventsByYear && (
```

Comment:
This is not quite correct as in case of `upcomingEventsByYear` is empty object condition will still be truthy. I'd suggest

```js
Object.keys(upcomingEventsByYear).length && ...
```